CliWrap/PipeSource.cs

```diff
@@ -130,4 +152,79 @@ await _command
             .WithStandardOutputPipe(PipeTarget.ToStream(destination))
             .ExecuteAsync(cancellationToken)
             .ConfigureAwait(false);
+}
+
+internal class StreamDelegatePipeSource : PipeSource
+{
+    private readonly Action<Stream> _source;
+
+    public StreamDelegatePipeSource(Action<Stream> source) => _source = source;
+
+    public override Task CopyToAsync(Stream destination, CancellationToken cancellationToken = default)
+    {
+        _source(destination);
+        return Task.CompletedTask;
+    }
+}
+
+internal class StreamAsyncDelegatePipeSource : PipeSource
+{
+    private readonly Func<Stream, CancellationToken, Task> _source;
+
+    public StreamAsyncDelegatePipeSource(Func<Stream, CancellationToken, Task> source) => _source = source;
+
+    public override async Task CopyToAsync(Stream destination, CancellationToken cancellationToken = default) =>
+        await _source(destination, cancellationToken).ConfigureAwait(false);
+}
```

Comment:
Actually, I realize now that what we did was define an API to creating a `PipeSource` from a delegate instead of by defining a new class.

Because of that, maybe it's better to call these `AnonymousPipeSource` and `AsyncAnonymousPipeSource`. And the factory method could be renamed from `PipeSource.FromDelegate(...)` to `PipeSource.Create(...)` to remove confusion because these are not equivalent to `PipeTarget.ToDelegate(...)` that we already have.

Reply:
Maybe we can also add a symmetrical `PipeTarget.Create(...)` too? You don't have to do it in this PR though, I can add it myself later.